### ISSUE TITLE
HomeKit - No Response

### ISSUE THREAD

**Issue by svh1985** [ikalchev/HAP-python#225]
Hi there,

I manually setup a machine with Debian Buster and installed Hass.io. After that, I connected my Home Assistant to HomeKit via my iPhone. The pairing was successful and the devices show-up in HomeKit.

All good until I noticed that the devices regularly had the status No Response. The devices keeps going between No Response and “Responding”.

**Some questions I have:**
- What can cause this?
- How can I debug this?
- Does the host system also need avahi-daemon installed?

**My config:**
- i7 16 GB 265GB SSD
- Debian GNU/Linux 10 (buster)
- Home Assistant Core v0.105.3
- Supervisor v200
- Connected via ethernet cable (no VLAN)
- Own domain with Lets Encrypt Certificate
- Opened port in router to HA on port 443 for HA
- Opened port in router to UDP 5353 and TCP 51827 for mDNS and HomeKit
- Apple TV 4 as HomeKit bridge at 172.16.0.16 (in log)

**Debug log HomeKit Home Assistant**
```
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 796788239, 'iid': 9, 'ev': True}, {'aid': 796788239, 'iid': 10, 'ev': True}]}
2020-02-11 15:13:33 INFO (Thread-57) [pyhap.hap_server] 172.16.0.12 - "GET /accessories HTTP/1.1" 200 -
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 796788239.9
2020-02-11 15:13:33 DEBUG (Thread-55) [pyhap.hap_server] Request PUT from address '('172.16.0.65', 62651)' for path '/characteristics'.
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 796788239.10
2020-02-11 15:13:33 DEBUG (Thread-55) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 3558740872, 'iid': 9, 'ev': True}]}
2020-02-11 15:13:33 INFO (Thread-54) [pyhap.hap_server] 172.16.0.102 - "PUT /characteristics HTTP/1.1" 204 -
2020-02-11 15:13:33 DEBUG (Thread-55) [pyhap.accessory_driver] Subscribed client ('172.16.0.65', 62651) to topic 3558740872.9
2020-02-11 15:13:33 INFO (Thread-55) [pyhap.hap_server] 172.16.0.65 - "PUT /characteristics HTTP/1.1" 204 -
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 3558740872, 'iid': 9, 'ev': True}]}
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 3558740872.9
2020-02-11 15:13:33 DEBUG (Thread-55) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 796788239, 'iid': 9, 'ev': True}, {'aid': 796788239, 'iid': 10, 'ev': True}]}
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 600182631, 'iid': 9, 'ev': True}]}
2020-02-11 15:13:33 DEBUG (Thread-55) [pyhap.accessory_driver] Subscribed client ('172.16.0.65', 62651) to topic 796788239.9
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 600182631.9
2020-02-11 15:13:33 DEBUG (Thread-55) [pyhap.accessory_driver] Subscribed client ('172.16.0.65', 62651) to topic 796788239.10
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 941295249, 'iid': 9, 'ev': True}]}
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 941295249.9
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 1011549895, 'iid': 9, 'ev': True}]}
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 1011549895.9
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 1187972988, 'iid': 9, 'ev': True}]}
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 1187972988.9
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 1282475955, 'iid': 9, 'ev': True}, {'aid': 1282475955, 'iid': 10, 'ev': True}, {'aid': 1282475955, 'iid': 11, 'ev': True}, {'aid': 1282475955, 'iid': 12, 'ev': True}]}
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 1282475955.9
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 1282475955.10
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 1282475955.11
2020-02-11 15:13:33 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 1282475955.12
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.hap_server] Request PUT from address '('172.16.0.16', 49615)' for path '/characteristics'.
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 1011549895, 'iid': 9, 'ev': True}]}
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 1011549895.9
2020-02-11 15:13:33 INFO (Thread-56) [pyhap.hap_server] 172.16.0.16 - "PUT /characteristics HTTP/1.1" 204 -
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 1282475955, 'iid': 9, 'ev': True}, {'aid': 1282475955, 'iid': 10, 'ev': True}, {'aid': 1282475955, 'iid': 11, 'ev': True}, {'aid': 1282475955, 'iid': 12, 'ev': True}]}
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 1282475955.9
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 1282475955.10
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 1282475955.11
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 1282475955.12
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 3558740872, 'iid': 9, 'ev': True}]}
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 3558740872.9
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 600182631, 'iid': 9, 'ev': True}]}
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 600182631.9
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 1187972988, 'iid': 9, 'ev': True}]}
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 1187972988.9
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 941295249, 'iid': 9, 'ev': True}]}
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 941295249.9
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 796788239, 'iid': 9, 'ev': True}, {'aid': 796788239, 'iid': 10, 'ev': True}]}
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 796788239.9
2020-02-11 15:13:33 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 796788239.10
2020-02-11 15:13:42 DEBUG (Thread-54) [pyhap.hap_server] Request PUT from address '('172.16.0.102', 62621)' for path '/characteristics'.
2020-02-11 15:13:42 DEBUG (Thread-54) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 1011549895, 'iid': 9, 'ev': False}]}
2020-02-11 15:13:42 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 1011549895.9
2020-02-11 15:13:42 INFO (Thread-54) [pyhap.hap_server] 172.16.0.102 - "PUT /characteristics HTTP/1.1" 204 -
2020-02-11 15:13:42 DEBUG (Thread-54) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 1282475955, 'iid': 9, 'ev': False}, {'aid': 1282475955, 'iid': 10, 'ev': False}, {'aid': 1282475955, 'iid': 11, 'ev': False}, {'aid': 1282475955, 'iid': 12, 'ev': False}]}
2020-02-11 15:13:42 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 1282475955.9
2020-02-11 15:13:42 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 1282475955.10
2020-02-11 15:13:42 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 1282475955.11
2020-02-11 15:13:42 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 1282475955.12
2020-02-11 15:13:42 DEBUG (Thread-54) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 600182631, 'iid': 9, 'ev': False}]}
2020-02-11 15:13:42 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 600182631.9
2020-02-11 15:13:42 DEBUG (Thread-54) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 1187972988, 'iid': 9, 'ev': False}]}
2020-02-11 15:13:42 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 1187972988.9
2020-02-11 15:13:42 DEBUG (Thread-54) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 941295249, 'iid': 9, 'ev': False}]}
2020-02-11 15:13:42 DEBUG (Thread-54) [pyhap.accessory_driver] Subscribed client ('172.16.0.102', 62621) to topic 941295249.9
2020-02-11 15:13:52 DEBUG (Thread-56) [pyhap.hap_server] Request PUT from address '('172.16.0.16', 49615)' for path '/characteristics'.
2020-02-11 15:13:52 DEBUG (Thread-56) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 1011549895, 'iid': 9, 'ev': False}]}
2020-02-11 15:13:52 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 1011549895.9
2020-02-11 15:13:52 INFO (Thread-56) [pyhap.hap_server] 172.16.0.16 - "PUT /characteristics HTTP/1.1" 204 -
2020-02-11 15:13:52 DEBUG (Thread-56) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 941295249, 'iid': 9, 'ev': False}]}
2020-02-11 15:13:52 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 941295249.9
2020-02-11 15:13:52 DEBUG (Thread-56) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 1187972988, 'iid': 9, 'ev': False}]}
2020-02-11 15:13:52 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 1187972988.9
2020-02-11 15:13:52 DEBUG (Thread-56) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 600182631, 'iid': 9, 'ev': False}]}
2020-02-11 15:13:52 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 600182631.9
2020-02-11 15:13:52 DEBUG (Thread-56) [pyhap.hap_server] Set characteristics content: {'characteristics': [{'aid': 1282475955, 'iid': 11, 'ev': False}, {'aid': 1282475955, 'iid': 12, 'ev': False}, {'aid': 1282475955, 'iid': 10, 'ev': False}, {'aid': 1282475955, 'iid': 9, 'ev': False}]}
2020-02-11 15:13:52 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 1282475955.11
2020-02-11 15:13:52 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 1282475955.12
2020-02-11 15:13:52 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 1282475955.10
2020-02-11 15:13:52 DEBUG (Thread-56) [pyhap.accessory_driver] Subscribed client ('172.16.0.16', 49615) to topic 1282475955.9
2020-02-11 15:14:15 DEBUG (Thread-57) [pyhap.hap_server] Cleaning connection to ('172.16.0.12', 49713)
2020-02-11 15:14:38 DEBUG (Thread-55) [pyhap.hap_server] Cleaning connection to ('172.16.0.65', 62651)
2020-02-11 15:14:38 DEBUG (Thread-56) [pyhap.hap_server] Cleaning connection to ('172.16.0.16', 49615)
2020-02-11 15:14:39 DEBUG (Thread-54) [pyhap.hap_server] Cleaning connection to ('172.16.0.102', 62621)

```

**Comment by Pythonaire** [comment]
"Hass.io" and "HAP-Python" are both HomeKit bridges. Both needs avahi to announce the service. 
If you like to connect a device with "HAP-Python", no hass.io is needed. The same vis versa.
Having both, the device is binded to HAP-Python AND hass.io,  sounds strange to me. That might be the described problem.

**Comment by svh1985** [comment]
I kept looking at my HA config and thinking that was the cause of the HomeKit No Response issues as I recently migrated to a new machine. But a comment on the HA forums about resetting the router turned my attention towards my router and it turned out, that for some reason, my DrayTek 2926 router was messing around with the multicast messages.

So, I disabled the following options in my router’s config and it started working again! 😄 

- IGMP Proxy
- IGMP Snooping
- IGMP Fast Leave
